ex1/functions.py

```diff
@@ -65,24 +73,57 @@ def f():
 # compare against. The function should return a list with equal length as the
 # input list containing for each number in the original list either True or
 # False if the number was greater than or equal to n.
+def check_list( a, n ):
+    ret = []
+    for i in a:
+        ret.append( i >= n )
+    return ret
 
 # Define a function named check_list_nth that does the same as check_list but
 # uses every nth element of the input list (including the first one). You will
 # need a third input argument.
+def check_list_nth( a, n, nth ):
+    ret = []
+    i = 0
```

Comment:
You can use enumerate to iterate through a list. Then you do not have to use a while loop and count manually.